WooCommerce/Classes/Tools/InfiniteScroll/PaginationTracker.swift

```diff
@@ -0,0 +1,152 @@
+import Yosemite
+
+/// Delegate of `PaginationTracker` that implements syncing per page number and size.
+protocol PaginationTrackerDelegate: class {
+    typealias SyncCompletion = (Result<Bool, Error>) -> Void
+
+    /// Syncs a page by the number and size, and returns an error or whether there might be the next page on completion.
+    ///
+    /// - Parameter reason: A value passed from `resync` or `syncFirstPage`. This can
+    ///                     be used to decide how to perform the sync.
+    /// - Parameter onCompletion: called on sync completion that returns either an error or a boolean that indicates
+    ///                           whether there might be the next page to sync.
+    func sync(pageNumber: Int, pageSize: Int, reason: String?, onCompletion: SyncCompletion?)
+}
+
+/// Keeps track of the pagination for API syncing to support infinite scroll.
+final class PaginationTracker {
```

Comment:
Should we mark `SyncCoordinator` as `@deprecated`?

Reply:
good idea, I did so in https://github.com/woocommerce/woocommerce-ios/pull/2750/commits/f2c848641884637de6fa106e10f92748ebb806e4 but now we have 23 issues in WooCommerce and 7 issues in WooCommerceTests from the usage of `SyncingCoordinator` - do you think it's fine we leave the warnings in the project and I can work on replacing `SyncingCoordinator` for hack week next week?

Reply:
I asked for opinions in CGPNUU63E-p1599574615070200-slack

Reply:
I reverted the deprecation in https://github.com/woocommerce/woocommerce-ios/pull/2750/commits/cd9406458dc8a13fdf4b8aa00beb027924cdc05f so that we have no build warnings!